### PR TITLE
feat: reply to original question

### PR DESCRIPTION
**Pull request by Sleinmaster**
## Description

If someone adds a bot command into a reply the bot replies to the same message, otherwise to the message initiating the bot reply.

## Test Results

Without multi plane support 
image

With Multi Plane support.
image
image
image

## Discord Username

Slein#8982

### CODE REVIEW

**Comment by pdellaert** [comment]
I love this idea! I do have a few improvements/suggestions:

You'll also have to add it here: https://github.com/flybywiresim/discord-bot/blob/412138d28386b5fefc49ce8015c04abe43b5daaf/src/lib/typeCommand.ts#L111 . This is for when the multi-aircraft support is enabled and there's different replies depending on the type of aircraft. (`.airframe` for instance)

As well as here: https://github.com/flybywiresim/discord-bot/blob/412138d28386b5fefc49ce8015c04abe43b5daaf/src/lib/typeCommand.ts#L126 . As this is the case when someone executes the aircraft specific version directly (`.a32nx airframe` for instance. 

To test this, you can set this to true: https://github.com/flybywiresim/discord-bot/blob/412138d28386b5fefc49ce8015c04abe43b5daaf/src/lib/typeCommand.ts#L9

**Comment by Sleinmaster** [comment]
@pdellaert 
Thanks for the suggestions.
I mainly created this pr as a reminder for myself since I could only watch the code and edit it on my phone. 
I planned to do it in the same style in all the places in that file, where a message is sent, as soon as I'm back home (next weekend).

Also I was thinking about the other cases, where the bot command is not in a reply. There I see the two possibilities to either just post the message, like now, or to reply to the message with the command in it. 
I might ask about that next week in the appropriate discord channel.

**Comment by pdellaert** [comment]
@Sleinmaster , sounds good! Yeah, wasn't sure whether it was the right time to already comment. But good you are looking at it. 

No rush on this, but it is a great idea!

For the regular `CommandDefinition`, it would mean an overhaul of executors everywhere, or to define a whole new logic and replace the existing executors, which will get a lot more complex. 

Instead, I think we should focus first of moving everything to `MessageCommandDefintion` where possible. As per my investigation when I created this model, there's 2/3rds of the commands that can become `MessageCommandDefinition`, and I think there might actually be another 10% that can move. 

BenW is working on that in #406

**Review by pdellaert — APPROVED** [review]
### Test Results

I have one remark at the end, otherwise tests

✅ Disable multi-type - multi-typed command - direct request
✅ Disable multi-type - single-typed command - replied request
Screenshot 2023-02-26 at 1 19 58 PM

✅ Enable multi-type - single-typed command - direct request
✅ Enable multi-type - multi-typed command - direct request with type a32nx
✅ Enable multi-type - multi-typed command - direct request with type a380x
Screenshot 2023-02-26 at 1 23 07 PM

✅ Enable multi-type - multi-typed command - replied request with type a32nx
✅ Enable multi-type - multi-typed command - replied request with type a380x
✅ Enable multi-type - single-typed command - replied request
Screenshot 2023-02-26 at 1 23 19 PM

✅ Enabled multi-type - multi-typed command - direct request with choice
Screenshot 2023-02-26 at 1 23 51 PM

✅ Enabled multi-type - multi-typed command - direct request with choice - After a32nx choice
Screenshot 2023-02-26 at 1 24 02 PM

✅ Enabled multi-type - multi-typed command - replied request with choice
Screenshot 2023-02-26 at 1 25 42 PM

✅ Enabled multi-type - multi-typed command - replied request with choice - CHOICE MADE BY COMMAND EXECUTOR
Screenshot 2023-02-26 at 1 25 57 PM

✅ Enabled multi-type - single-typed command - direct request
✅ Enabled multi-type - single-typed command - replied request
Screenshot 2023-02-26 at 1 26 43 PM

### Note on replied behavior and choice request

Currently, the `typeCommand`, when presenting a choice waits for a reaction from the _command executor_. This means that when replying to a request, and the choice is presented, the user that executes the command _must_ select the type to show the output. 

This is not a limitation of this PR, it has always been there, so this does not block this PR. I can fix this in the future by digging a bit deeper and also allowing the person who it was relied to, to react. 

Sequence to show behavior:

**Step 1** - User requests help, other user responds, choice is presented
Screenshot 2023-02-26 at 1 38 53 PM

**Step 2** - User selects choice, but is not given a reply
Screenshot 2023-02-26 at 1 39 00 PM

**Step 3** - Other user selects A32, choice is locked and message replaced (with proper reply)
Screenshot 2023-02-26 at 1 39 10 PM

**Comment by Sleinmaster** [comment]
> Currently, the typeCommand, when presenting a choice waits for a reaction from the command executor. This means that when replying to a request, and the choice is presented, the user that executes the command must select the type to show the output.
> 
> This is not a limitation of this PR, it has always been there, so this does not block this PR. I can fix this in the future by digging a bit deeper and also allowing the person who it was relied to, to react.

That I can add fairly easy. I will change that quickly when I have time in this PR.

**Comment by benw202** [comment]
Just going to ping the CS team with this before it is merged to make sure they are happy as this will change how the bot reacts/might be used in support.

**Comment by Sleinmaster** [comment]
Screenshot 2023-03-04 131443
Screenshot 2023-03-04 131645

For the added user ID's (only if not replied to executor)

File: src/lib/command.ts
Comment (benw202) [inline src/lib/command.ts:147]:
```suggestion
            embed.setFooter({ text: `${existingFooterText} Executed by ${msg.author.tag} - ${msg.author.id}` });
```

**Comment by benw202** [comment]
@pdellaert - have re-requested your review on this as you've been more involved :)

File: src/lib/typeCommand.ts
Comment (pdellaert) [inline src/lib/typeCommand.ts:2]:
```suggestion
import { CommandDefinition, isMessageCommand, MessageCommandDefinition, hasRequiredPermissions, sendPermissionsEmbed, replyToCommandOrQuestion } from './command';
```
`replyToMessage` no longer exists and will cause a compile error.

**Review by pdellaert — CHANGES_REQUESTED** [review]
There's a bug ;). Will test again once fixed.

**Review by pdellaert — COMMENTED** [review]
Looks good, will do another test run and approve once done

**Comment by pdellaert** [comment]
### Test Results

I have one remark at the end which is not of any impact on this PR, otherwise tests are all good!

✅ Disable multi-type - multi-typed command - direct request
✅ Disable multi-type - single-typed command - replied request
Screenshot 2023-03-05 at 9 22 02 PM

✅ Enable multi-type - single-typed command - direct request
✅ Enable multi-type - multi-typed command - direct request with type a32nx
✅ Enable multi-type - multi-typed command - direct request with type a380x
Screenshot 2023-03-05 at 9 23 53 PM

✅ Enable multi-type - multi-typed command - replied request with type a32nx
✅ Enable multi-type - multi-typed command - replied request with type a380x
Screenshot 2023-03-05 at 9 26 48 PM

✅ Enable multi-type - single-typed command - replied request
Screenshot 2023-03-05 at 9 27 13 PM

✅ Enabled multi-type - multi-typed command - direct request with choice
Screenshot 2023-03-05 at 9 30 31 PM

❌❌❌❌❌❌❌❌❌❌
❌ Enabled multi-type - multi-typed command - direct request with choice - After a32nx choice made by original user
Screenshot 2023-03-05 at 9 30 44 PM

Failure: Multiple "executed by"

❌❌❌❌❌❌❌❌❌❌
❌ Enabled multi-type - multi-typed command - replied request with choice - Choice made by Command executor
Screenshot 2023-03-05 at 9 33 27 PM

Failure: Multiple "executed by"

❌❌❌❌❌❌❌❌❌❌
❌ Enabled multi-type - single-typed command - direct request
❌ Enabled multi-type - single-typed command - replied request
Screenshot 2023-03-05 at 9 35 39 PM

Failure: Multiple "executed by"

### Bug

Interesting bug with the multiple footers. The reason seems that the existingFooterText variable remains set across command executions, then the catch doesn't clear it, so it kept an existing footer and re-used it. I have a proposal to fix this with some small enhancements as well.

File: src/lib/command.ts
Comment (pdellaert) [inline src/lib/command.ts:151]:
```suggestion
export async function replyToCommandOrQuestion(msg: Message, embed: EmbedBuilder) : Promise<Message<boolean>> {
    let existingFooterText = '';
    return msg.fetchReference()
        .then((res) => {
            const existingFooter = embed.data.footer;
            if (existingFooter) {
                existingFooterText = existingFooter.text;
            }
            embed.setFooter({ text: `${existingFooterText}\n\nExecuted by ${msg.author.tag} - ${msg.author.id}` });
            return res.reply({ embeds: [embed] });
        })
        .catch(() => msg.reply({ embeds: [embed] }));
}
```

This fixes the bug in my last testing and cleans up the text a bit as well (no unnecessary space)

**Review by pdellaert — CHANGES_REQUESTED** [review]
Fixing the bug found during testing

File: src/lib/command.ts
Comment (pdellaert) [inline src/lib/command.ts:147]: Sorry about previous posts, brainfart. Here's the issue, you update the embed in place, which updates the embed system wide with this footer. So every time this is executed, it adds a new line 

Examples:
Screenshot 2023-03-05 at 11 27 16 PM
Screenshot 2023-03-05 at 11 27 23 PM
Screenshot 2023-03-05 at 11 27 41 PM

**Review by pdellaert — CHANGES_REQUESTED** [review]

**Review by benw202 — APPROVED** [review]
LGTM! Thanks for the addition :)

**Pull request merged**
